frontend/src/services/share.service.ts

```diff
@@ -92,20 +94,25 @@ const createReverseShare = async (
   shareExpiration: string,
   maxShareSize: number,
   maxUseCount: number,
-  sendEmailNotification: boolean
+  sendEmailNotification: boolean,
+  sharesOptions?: ReverseShareOptions
 ) => {
   return (
     await api.post("reverseShares", {
       shareExpiration,
       maxShareSize: maxShareSize.toString(),
       maxUseCount,
       sendEmailNotification,
+      ...sharesOptions,
     })
   ).data;
 };
 
 const getMyReverseShares = async (): Promise<MyReverseShare[]> => {
-  return (await api.get("reverseShares")).data;
+  const shares = (await api.get("reverseShares")).data;
```

Comment:
`sharesOptions` can only be null if the reverse share was created before this update, or I'm wrong? If yes, I would make the migration directly in the databases instead of the frontend.

Reply:
I know, but the migration is not able to create the required UUID, maybe we should rather create it in the backend ?

Reply:
You're right it's a pain to create migrations in SQL for a SQLite db. But we should try to stick to this method as it handles the whole migration process really well. If we want to do it in the backend we would have to write the logic by our own.

I found an awful way to generate an UUID but it should work:
```sql
SELECT
  lower(
    hex(randomblob(4)) || '-' || hex(randomblob(2)) || '-' || '4' ||
    substr(hex( randomblob(2)), 2) || '-' ||
    substr('AB89', 1 + (abs(random()) % 4) , 1)  ||
    substr(hex(randomblob(2)), 2) || '-' ||
    hex(randomblob(6))
  ) id

```
[From Stackoverflow](https://stackoverflow.com/questions/66625085/sqlite-generate-guid-uuid-on-select-into-statement)

Reply:
I think we would need to add this into the migration, I'll test it right now : 
```sql
INSERT INTO "ReverseShareOptions" ("id", "shareId", "easyMode", "customLinkEnabled", "passwordEnabled", "descriptionEnabled", "maximalViewsEnabled")
SELECT
    lower(
        hex(randomblob(4)) || '-' || hex(randomblob(2)) || '-' || '4' ||
        substr(hex( randomblob(2)), 2) || '-' ||
        substr('AB89', 1 + (abs(random()) % 4) , 1)  ||
        substr(hex(randomblob(2)), 2) || '-' ||
        hex(randomblob(6))
    ),
    id, 0, 1, 1, 1, 1 FROM "ReverseShare";
```

Reply:
@stonith404, in fact I have been a bit busy, so I won't be able to finish this right now. I'll do it later or if you want to go on with this, but I'll finish it anyway.

Reply:
Yeah sure no stress :)